server/monitor-types/tailscale-ping.js

```diff
@@ -0,0 +1,89 @@
+// TODO: Handle Case 5
+// In cases where offline peers are not pingable, how should he handle timeout messages? Do we break out of the "ping (ip) timed out" loop and let the server re-call the check function? or do we include the "timed out" string in the expected error messages line and throw the error? (along with "no matching peer" and "is local Tailscale IP")
+
+const { MonitorType } = require("./monitor-type");
+const { UP } = require("../../src/util");
+
+// child_process
+const exec = require('child_process').exec;
+
+// cases accounted for
+// 1 calling tailscale ping to local
+// CMD: tailscale ping 100.100.100.1
+// 100.100.100.1 is local Tailscale IP
+// 2 the normal scenario
+// CMD: tailscale ping 100.100.100.2
+// pong from (hostname) (100.100.100.2) via [2001:1245:1234:1234::abc]:12345 in 123ms
+// 3 ping a nonexistent IP or inaccessible node
+// CMD: tailscale ping 100.100.100.3
+// no matching peer
+// 4 ping a peer that you cannot make a outgoing connection to
+// CMD: tailscale ping 100.100.100.4
+// multiple 'pong from' lines (pong from hostname (100.100.100.4) via DERP(tok) in 123ms) then ends with a 'direct connection not established' line
+// We only care about the first 'pong from' line for the purpose of checking uptime. (this is a good example use of "tailscale ping" becasue traditional ICMP pings would not be able to check the uptime of peers if outgoing connections to that specified peer are blocked)
+// 5 pinging a offline peer
+// CMD: tailscale ping 100.100.100.5
+// multiple 'timed out' lines (ping "100.100.100.5" timed out) then ends with a 'no reply' line
+// This needs to be worked on
```

Comment:
I am going to be controversioal:
This comment is really hard to parse and in general, large comments explaining functionality rott realy fast.
Could you integrate this into the code where you account for these cases if they are not self-explanatory?

See https://www.youtube.com/watch?v=Bf7vDBBOBUA for a better explaination of this argument

Reply:
Understood, this is the first time I've written and contributed code in this fashion, I'll learn as I go, and I appreciate your suggestions.